### DISCUSSION TITLE
time zone is not taken into account in run_id

### DISCUSSION THREAD

**Discussion by dgabidullin** [apache/airflow#27471] [Q&A] [answered]
version: v2.3.4

Hi airflow community!
In the run_id field, part with the logical_date is generated in utc but we running our dags with default_timezone=Europe/Moscow in conf. Can we somehow specify when generating run_id part with logical_date should be in default_timezone?

![run_id](https://user-images.githubusercontent.com/43440135/199525157-2fa71fde-ee87-4d65-b4ca-2b7881e482f2.png)

**Comment by potiuk** [comment, marked as answer]
Not currently - but feel free to provide PR or open an issue.

**Comment by studB** [comment]
For that, You should modify code like below.
file : airflow/utils/timezone.py, airflow2.5.3v 

```
def coerce_datetime(v: dt.datetime | None, tz: dt.tzinfo | None = None) -> DateTime | None:
    if v is None:
        return None
    if tz is None:
        from airflow.settings import TIMEZONE
        tz = TIMEZONE
    if isinstance(v, DateTime):
        return v.astimezone(tz) if v.tzinfo else make_aware(v, tz).astimezone(tz)
    # Only dt.datetime is left here.
    return pendulum.instance(v if v.tzinfo else make_aware(v, tz)).astimezone(tz)
```
